src/transformers/models/udop/tokenization_udop.py

```diff
@@ -0,0 +1,1506 @@
+        """
+        token_ids_0 = self._add_eos_if_not_present(token_ids_0)
+        if token_ids_1 is None:
+            return token_ids_0
+        else:
+            token_ids_1 = self._add_eos_if_not_present(token_ids_1)
+            return token_ids_0 + token_ids_1
+
+    # Copied from transformers.models.t5.tokenization_t5.T5Tokenizer.__getstate__
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        state["sp_model"] = None
+        return state
+
+    # Copied from transformers.models.t5.tokenization_t5.T5Tokenizer.__setstate__
+    def __setstate__(self, d):
+        self.__dict__ = d
+
+        # for backward compatibility
+        if not hasattr(self, "sp_model_kwargs"):
+            self.sp_model_kwargs = {}
+
+        self.sp_model = spm.SentencePieceProcessor(**self.sp_model_kwargs)
+        self.sp_model.Load(self.vocab_file)
+
+    # Copied from transformers.models.t5.tokenization_t5.T5Tokenizer._tokenize
+    def _tokenize(self, text: str) -> List[str]:
+        """Take as input a string and return a list of strings (tokens) for words/sub-words"""
+        return self.sp_model.encode(text, out_type=str)
+
+    def _convert_token_to_id(self, token):
+        """Converts a token (str) in an id using the vocab."""
+        if token.startswith("<extra_id_"):
+            match = re.match(r"<extra_id_(\d+)>", token)
+            num = int(match.group(1))
+            return self.vocab_size - num - 1 - self._other_extra_ids - self._loc_extra_ids - self._extra_ids * 4
+        elif token.startswith("<extra_l_id_"):
+            match = re.match(r"<extra_l_id_(\d+)>", token)
+            num = int(match.group(1))
+            return self.vocab_size - num - 1 - self._other_extra_ids - self._loc_extra_ids - self._extra_ids * 3
+        elif token.startswith("</extra_l_id_"):
+            match = re.match(r"</extra_l_id_(\d+)>", token)
+            num = int(match.group(1))
+            return self.vocab_size - num - 1 - self._other_extra_ids - self._loc_extra_ids - self._extra_ids * 2
+        elif token.startswith("<extra_t_id_"):
+            match = re.match(r"<extra_t_id_(\d+)>", token)
+            num = int(match.group(1))
+            return self.vocab_size - num - 1 - self._other_extra_ids - self._loc_extra_ids - self._extra_ids
+        elif token.startswith("</extra_t_id_"):
+            match = re.match(r"</extra_t_id_(\d+)>", token)
+            num = int(match.group(1))
+            return self.vocab_size - num - 1 - self._other_extra_ids - self._loc_extra_ids
+        elif token.startswith("<loc_"):
+            match = re.match(r"<loc_(\d+)>", token)
+            num = int(match.group(1))
+            return self.vocab_size - num - 1 - self._other_extra_ids
+        elif token.startswith("<other_"):
+            match = re.match(r"<other_(\d+)>", token)
+            num = int(match.group(1))
+            return self.vocab_size - num - 1
```

Comment:
@younesbelkada , about the bug concerning the loading of the slow tokenizer, I think the problem comes from those lines. 

I think this logic is contradictory with adding `<extra_l_id_Xxx>` tokens as added tokens, which is the case because they are listed inside https://huggingface.co/nielsr/udop-large/blob/main/added_tokens.json .


----
For context, the problematic snippet:
```python
from transformers import UdopTokenizer

tokenizer = UdopTokenizer.from_pretrained("nielsr/udop-large")
```
which returns 
```
Traceback (most recent call last):
  File "/home/lucile/tmp/test_tok.py", line 3, in <module>
    tokenizer = UdopTokenizer.from_pretrained("nielsr/udop-large")
  File "/home/lucile/repos/transformers/src/transformers/tokenization_utils_base.py", line 1812, in from_pretrained
    return cls._from_pretrained(
  File "/home/lucile/repos/transformers/src/transformers/tokenization_utils_base.py", line 2024, in _from_pretrained
    raise ValueError(
ValueError: Wrong index found for <extra_l_id_0>: should be 32199 but found 32100.
```